node/network/collator-protocol/src/collator_side.rs

```diff
@@ -195,6 +195,18 @@ where
 
 	state.our_validators_groups.insert(relay_parent, our_validators.clone());
 
+	// We may be already connected to some of the validators. In that case,
+	// advertise a collation to them right away.
+	for validator in our_validators.iter() {
+		if let Some(peer) = state.known_validators.get(&validator).cloned() {
+			if let Some(view) = state.peer_views.get(&peer) {
+				if view.contains(&relay_parent) {
+					advertise_collation(ctx, state, relay_parent, vec![peer]).await?;
```

Comment:
```suggestion
		if let Some(peer) = state.known_validators.get(&validator) {
			if let Some(view) = state.peer_views.get(peer) {
				if view.contains(&relay_parent) {
					advertise_collation(ctx, state, relay_parent, vec![peer.clone()]).await?;
```

Reply:
```
warning: cannot borrow `*state` as mutable because it is also borrowed as immutable
   --> node/network/collator-protocol/src/collator_side.rs:204:31
    |
201 |         if let Some(peer) = state.known_validators.get(&validator) {
    |                             ---------------------- immutable borrow occurs here
...
204 |                     advertise_collation(ctx, state, relay_parent, vec![peer.clone()]).await?;
    |                                              ^^^^^                     ---- immutable borrow later used here
    |                                              |
    |                                              mutable borrow occurs here
    |
    = note: `#[warn(mutable_borrow_reservation_conflict)]` on by default
    = warning: this borrowing pattern was not meant to be accepted, and may become a hard error in the future
    = note: for more information, see issue #59159 <https://github.com/rust-lang/rust/issues/59159>
```
Am i doing it wrong?

Reply:
Edited the suggestion.